content/tools-and-utilities/self-assignable-roles.md

```diff
@@ -0,0 +1,213 @@
+
+The group mode and other restrictions from the role group and role still apply to the roles in the role menu.\
+\
+After you type in the command, you will be taken through the setup process. If you want to disable DMs, create a custom message, or add new role to your role menu, be sure to read until the end.
+
+### Step by step tutorial
+
+{{% notice warning %}}
+
+Make sure you created your [role commands](#roles) and assigned them a [role group](#role-groups) before starting. Role menu will **NOT** work if you have not done so. All switches and flags (nodm, rr, etc...) start with hyphen symbol `-`, not your prefix.
+
+{{% /notice %}}
+
+Once you've made your role commands and assigned them to a role group,\
+go to the channel in Discord where you want the role menu to be created. There you type `-rolemenu create (role group name)` (in this example, the role group name is "Sports").
+
+![Rolemenu Create command](rolemenu_create.png)
+
+As you can see, the bot started creating the menu, and has asked you to add the emoji for the first role, **Soccer**. I'll react on the message with my emoji of choice:&#x20;
+
+![First reaction on a rolemenu setup](rolemenu_react.png)
+
+Now the bot wants you to react with the emoji for the second role, **Hockey:**
+
+![Second reaction on a rolemenu setup](rolemenu_react_2.png)
+
+And similar for the final role, **Basketball:**
+
+![Third reaction on a rolemenu setup](rolemenu_react_3.png)
+
+And we're done — people can start using the menu by clicking on the reaction associated with their desired role. The setup message will be automatically deleted in a couple of minutes, but you can also delete it manually if you so desire.
+
+### Custom message
+
+To create a custom message for your role menu like event role menu you saw above, you will need to send a message. Then get the ID of the message by following the steps in [how to get a message ID](self-assignable-roles.md#how-to-get-a-message-id-desktop) and type in the following command, `-rolemenu create (group name) -m (message id)`. You would then complete the role menu like you would any normal role menu&#x20;
+
+### Disabling DM confirmation
+
+If you do not want the bot to send you a DM when you are given or removed from a role, type in the following command `-rolemenu update (message id) -nodm`&#x20;
+
+After you have finish editing or creating your role menu, it will display whether DM notifications are enabled or not.
+
+![Rolemenu flags display](rolemenu_flags.png)
+
+{{% notice warning %}}
+
+Note that YAGPDB does not allow you to disable warning DMs such as cool-down messages with the `nodm` switch or any other method.
+
+{{% /notice %}}
+
+### Remove roles on reaction remove
+
+Remove roles on reaction remove, this means instead of the old toggling mode, adding reactions will strictly give you the role, and removing reactions will only take away the role from you. **This mode will be on by default for new menus.**&#x20;
```

Comment:
```suggestion
Remove roles on reaction remove, this means instead of the old toggling mode, adding reactions will strictly give you the role, and removing reactions will only take away the role from you. This mode is on by default for new menus.
```